ldk/javascript/src/whisper/form.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Components, Form, Whisper, WhisperComponentType } from '.';
+
+export class LdkForm {
+    children: Array<Components>;
+
+    componentState: Map<string, any>;
+
+    constructor(children: Array<Components>) {
+        this.componentState = new Map<string, any>();
+        this.children = children;
+
+        this.setInitialComponentState(children);
+        this.registerListeners(children);
+    }
+
+    getComponentState(): Map<string, any> {
+        return this.componentState;
+    }
+
+    private isOnChangeInput(child: any): boolean {
+        return (child.onChange !== undefined && child.name !== undefined);
+    }
+
+    private isOnSelectInput(child: any): boolean {
+        return (child.onSelect !== undefined && child.name !== undefined);
+    }
+
+    private setInitialComponentState(children: Array<any>) {
+        children.forEach((child: any) => {
```

Comment:
do you need to check for children != undefined?

Reply:
I added a test to make sure of behavior with empty children collection.

Reply:
I will add another test for undefined.

Reply:
probably could express this as:
```suggestion
        children?.forEach((child: any) => {
```